src/classes/services/CaseService.cls

```diff
@@ -60,4 +65,57 @@ public inherited sharing class CaseService {
         update casesToUpdate;
         insert appointmentResources;
     }
+
+    public static void createTimeSlot(List<Case> newCases) {
+
+        List<Time_Slot__c> timeSlots = new List<Time_Slot__c>();
+
+        Map<String, Operating_Hours__c> operatingHoursByIdsAndDates = new Map<String, Operating_Hours__c>();
+
+        Set<Id> accIds = new Set<Id>();
+
+        for (Case caseRecord : newCases) {
+            accIds.add(caseRecord.AccountId);
+        }
+
+        for (Account account : AccountSelector.getAccountByIds(accIds)) {
+            for (Operating_Hours__c oh : account.Operating_Hours__r) {
+                operatingHoursByIdsAndDates.put(String.valueOf(account.Id) + String.valueOf(oh.Date__c), oh);
+            }
+        }
+
+        for (Case caseRecord : newCases) {
+            Operating_Hours__c operatingHours = operatingHoursByIdsAndDates.get(
+                    String.valueOf(caseRecord.AccountId) + String.valueOf(caseRecord.Appointment_Date__c));
+
+            if (isTimeSuitable(caseRecord, operatingHours)) {
+                timeSlots.add(new Time_Slot__c(Start__c = caseRecord.Start_Time__c, Operating_Hours__c = operatingHours.Id));
+            } else {
+                caseRecord.addError('Proposed time is outside of boutique`s working hours. Please provide time within '
+                        + operatingHours.Start_Time__c.hour() + ':' + operatingHours.Start_Time__c.minute() +
+                        '-' + operatingHours.End_Time__c.hour() + ':' + operatingHours.End_Time__c.minute());
+            }
+        }
+
+        insert timeSlots;
```

Comment:
MAJOR
Instead of making another method in trigger to create Appointment Resources you can create them here after time slots as you already have these slots.

`        for (TimeSlot__c timeSlot : timeSlots) {
            appointmentResources.add(new AppointmentResource__c(
                    Resource__c = defaultBoutiqueResource.Id,
                    TimeSlot__c = timeSlot?.Id,
                    Appointment__c = filteredCasesByIds.get(timeSlot.CaseId__c)?.Id
            ));
        }
        insert appointmentResources;`